### ISSUE TITLE
Generate a diagram from markdown

### ISSUE THREAD

**Issue by RobOK2050** [zsviczian/obsidian-excalidraw-plugin#550]
User would like to have an Excalidraw diagram be generated from headings and bullets from a markdown file. (Much like Mind Node can import markdown). Each Heading 1 goes in a big box. Each Heading 2 is a slightly smaller box connected with a line. Text is shown inside box or on hover.

Would like to be able to edit the underlying markdown not just read only.

There is a mind map plugin for Obsidian, but I like the look of Excalidraw better.

Not sure if this fits in with your vision of the plug in.

**Comment by zsviczian** [comment]
something like this?
https://zsviczian.github.io/obsidian-excalidraw-plugin/Examples/templater_mindmap.html

**Comment by RobOK2050** [comment]
Yes, that is on the right track. I think what i have in mind might be beyond what is possible. I am envisioning where you could easily switch between editing the outline and editing the diagram. And that some snippet of the text shows on the diagram but you could click or hover to see more.  You could rearrange the outline and the graphic would be updated. I think the MindMap extension can do this, but excalidraw has much more graphical ability, it would be nice to have both

**Comment by kazdonkai** [comment]
I would love to have this feature too, but @zsviczian's example seems to require running a script to convert it.
I haven't actually tried it, so I'm imagining.

If possible, I'd like to see the mindmap generated dynamically when I edit the markdown notes in excalidraw.

In that case, it seems difficult to generate from bullet points.
So, would it be feasible to create a node for each heading hierarchy (number of #'s) with a box type (color, size, etc.), put them in order, and join them with an edge?

Isn't @RobOK2050's request similar to the above?

**Comment by dredhorse** [comment]
I get a 
```
plugin:templater-obsidian:82 Templater Error: Template parsing error, aborting. 
 Cannot read properties of null (reading '1')
``` 
when I want to run the template on a note.
